lib/utils/validations.js

```diff
@@ -1,3 +1,38 @@
+import Joi from 'joi'
+import { payloadSchema } from './schema'
+import getEntityName from 'contentful-batch-libs/dist/utils/get-entity-name'
+
+const attachEntityName = (details, payload) => {
+  details.map((detail) => {
+    if (detail.path.length >= 2) {
+      detail.entity = getEntityName(payload[detail.path[0]][detail.path[1]])
+    }
+    return detail
+  })
+}
+const countInvalidEntities = (validationData) => {
+  const entityCount = validationData.reduce((entities, currentDetail) => {
+    if (!entities[currentDetail.path[0]]) {
+      entities[currentDetail.path[0]] = 1
+    } else {
+      entities[currentDetail.path[0]]++
+    }
+    return entities
+  }, {})
+  return Object.keys(entityCount).map((key) => `${key}:${entityCount[key]}`)
+}
+
+const assertPayload = (payload) => {
+  const result = Joi.validate(payload, payloadSchema, {allowUnknown: true, abortEarly: false})
+  if (result.error) {
+    attachEntityName(result.error.details, payload)
+    const invalidEntityCount = countInvalidEntities(result.error.details).join('\n')
```

Comment:
Lets join via comma + space instead of new line since we have only one line available in the Listr task error display.